app/controllers/concerns/validate_jsonapi_type.rb

```diff
@@ -0,0 +1,17 @@
+module ValidateJsonapiType
+  extend ActiveSupport::Concern
+
+  def validate_jsonapi_type(params, type)
+    # jsonapi-rb doesn't appear to validate the type of data record that's
+    # been given us, so even though we say "require(:session)" it won't
+    # validate the type field of the data object that was passed through.
+    #
+    # So for now, we do this by hand.
+    sent_type = params[:_jsonapi][:data][:type]
+    unless sent_type == type
+      raise ActionController::BadRequest.new(
+        "data type '#{sent_type}' unsupported'"
```

Comment:
may add expected type as well, thinking of when we need to scan thro error logs

Reply:
`"data type expected #{type} but received data type '#{sent_type}' unsupported'"`

Reply:
Sounds good!